src/main/java/org/folio/linked/data/integration/kafka/sender/search/AuthorityCreateMessageSender.java

```diff
@@ -35,18 +38,33 @@ public class AuthorityCreateMessageSender implements CreateMessageSender {
 
   @Override
   public Collection<Resource> apply(Resource resource) {
-    return Stream.of(resource)
-      .filter(this::test)
+    if (isAuthority(resource)) {
+      if (isNull(resource.getIndexDate())) {
+        return singletonList(resource);
+      }
+      return emptyList();
+    }
+    return findNotIndexedAuthoritiesInResource(resource);
+  }
+
+  private Collection<Resource> findNotIndexedAuthoritiesInResource(Resource resource) {
```

Comment:
what if authority level if lower then the next one?

resource -> outgoingEdge -> target - > outgoing Edge->  target(authority)

Reply:
fixed.